### PR TITLE
Bonds v1.5

### PR DESCRIPTION
**Pull request by 0xAurelius**
Once we roll this out we should merge the changes to main

### CODE REVIEW

File: contracts/inverse-bonds/types/Pro/Viewer.sol [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+            Terms memory term = terms[ids[i]];
+
+            uint256 expiry = term.fixedTerm ? block.timestamp + term.vesting : term.vesting;
+            require(expiry <= maxExpiry, "Bad expiry");
+
+            if (minAmountOut > markets[ids[i]].maxPayout) {
+                payouts[n] = payoutFor(amountIn, ids[i]);
+            } else {
+                payouts[n] = 0;
+            }
```
Comment (foudufafa) [inline contracts/inverse-bonds/types/Pro/Viewer.sol:213]: This might need more eyes on this but I believe this whole `if/else` is wrong. Lets assume this inverse bond is `KLIMA/USDC` thus we put in `KLIMA` and we get `USDC`. The `minAmountOut` here is in `USDC` as well as the `maxPayout` and so lets take an example.

If `minAmountOut = 100` and `maxPayout = 50`, then the we set the `payouts[n]` to the payout but `payoutFor()` function doesn't cap the value onto capacity of 

If the`minAmountOut` is **above** the `maxPayout`, then it seems to me that it should be set to `0` as in that market won't be able to satisfy the minimum required.

If the`minAmountOut` is **below** the `maxPayout`, then that should be `payoutFor()` used.

In other words, this seems reverse?

File: contracts/inverse-bonds/types/Pro/Viewer.sol [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            n++;
+        }
+        uint256 highestOut;
+        for (uint256 i; i < payouts.length; i++) {
+            if (payouts[i] > highestOut) {
+                highestOut = payouts[i];
+                id = ids[i];
+            }
+        }
+    }
```
Comment (foudufafa) [inline contracts/inverse-bonds/types/Pro/Viewer.sol:223]: There is a second problem I think for this function is that it returns the maximum payout for an inverse bond using `payoutFor()` which **ignores** the bond capacity and so it is entirely possible that this function returns market(s) that have no capacity (or lower than `minAmountOut`) which is wrong considering that the "min amount" requirement should be honored.

File: contracts/inverse-bonds/types/Pro/Viewer.sol [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+        return terms[_id].controlVariable - decay;
+    }
+
+    /**
+     * @notice             returns maximum quote token in for market
+   */
+    function maxIn(uint256 _id) public view returns (uint256) {
+        Metadata memory meta = metadata[_id];
+        return
+        markets[_id].maxPayout
```
Comment (foudufafa) [inline contracts/inverse-bonds/types/Pro/Viewer.sol:94]: This is also something I wonder if the `maxCapacity` should be considered as well because if the capacity goes down to something lower than `maxPayout`, what this returns is true for the `maxPayout` but not true for the "market" itself.

Maybe something like:

```
Math.min(
  markets[_id].maxPayout,
  markets[_id].maxCapacity,
)
```
